external/js/cothority/src/byzcoin/byzcoin-rpc.ts

```diff
@@ -168,6 +188,70 @@ export default class ByzCoinRPC implements ICounterUpdater {
         return header.version;
     }
 
+    /**
+     * Returns an observable proof for an instance that will be
+     * automatically updated whenever the instance is changed.
+     * It returns a BehaviorSubject, which is a special observable that
+     * keeps the current value.
+     * It also caches all requests, so if there are two requests for the
+     * same id, no new observable will be created.
+     * It connects to `getNewBlocks` to be informed whenever a new block is
+     * created.
+     * @param id of the instance to return
+     * @throws an error if the instance does not exist
+     */
+    async proofObservable(id: InstanceID): Promise<BehaviorSubject<Proof>> {
```

Comment:
Just a reflexion: "proofObservable" is a bit abstract and when you think about it, what the user really wants to observe is an instance, not the proof itself right? Also when I think about a "proof", in my mind that refers to something fixed on the blockchain, and not something that evolves and gets updated. Would changing this to "instanceObservale" make sense?

Reply:
absolutely - naming is hard. It started out as `instanceObservable`, in fact. But then, as it returned a proof, I renamed it to `proofObservable`. And `instanceObservableByReturningProofs` is a bit long ;)

Reply:
IMO its easier for a user to understand that "we observe an instance, which returns a proof because that's how we reference instances in Byzcoin", rather than "we observe a proof, because that's the result of observing an instance, therefore we observe an instance".